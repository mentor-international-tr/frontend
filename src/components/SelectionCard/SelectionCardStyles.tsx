import styled from "styled-components";
import vars from "lib/styles/variables";
const SelectionCard = styled.button`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1em;
  border: 2px solid ${vars.black};
  border-radius: 5px;
  width: 100%;
  margin: 0;
  cursor: pointer;
  &:hover {
    -webkit-box-shadow: 10px 10px 10px -2px rgba(0, 0, 0, 0.28);
    -moz-box-shadow: 10px 10px 10px -2px rgba(0, 0, 0, 0.28);
    box-shadow: 10px 10px 10px -2px rgba(0, 0, 0, 0.28);
  }
  &:focus {
    border: 5px solid ${vars.black};
  }
  &:active {
    border: 5px solid ${vars.blue};
  }
`;
const IconContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  border: 2px solid ${vars.black};
  padding: 1em;
`;

export { SelectionCard, IconContainer };
