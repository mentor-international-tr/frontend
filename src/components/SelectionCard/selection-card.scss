@import "../../lib/styles/variables.scss";

  .selection-card {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 1em;
    border: 2px solid $black;
    border-radius: 5px;
    width: 100%;
    margin:0;
    cursor: pointer;
    &:hover {
      -webkit-box-shadow: 10px 10px 10px -2px rgba(0, 0, 0, 0.28);
      -moz-box-shadow: 10px 10px 10px -2px rgba(0, 0, 0, 0.28);
      box-shadow: 10px 10px 10px -2px rgba(0, 0, 0, 0.28);
    }
    &:focus {
      border: 5px solid $black;       
    }
    &:active {
      border: 5px solid $blue;
    }
  }
  
  .selection-card__icon-container {
    display: flex;
    justify-content: center;
    align-items: center;
    border: 2px solid $black;
    padding: 1em;
  }
  