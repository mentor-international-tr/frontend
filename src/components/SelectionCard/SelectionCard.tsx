import React from "react";
import "./selection-card.scss";
export interface SelectionCardProps {
  iconElement: React.ReactElement;
  description: string;
  extraInfo: string;
  className?: string;
}

const SelectionCard: React.SFC<SelectionCardProps> = ({
  iconElement,
  description,
  extraInfo,
}) => {
  return (
    <div className="selection-card">
      <div className="selection-card__icon-container">
        {React.cloneElement(iconElement)}
      </div>
      {description ? (
        <h2 className="selection-card__description">
          {description}
        </h2>
      ) : null}
      {extraInfo ? (
        <p className="selection-card__extra-info">
          {extraInfo}
        </p>
      ) : null}
    </div>
  );
};

export default SelectionCard;
