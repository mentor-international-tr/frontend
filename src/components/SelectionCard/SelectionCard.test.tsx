import React from "react";
import { render, cleanup } from "react-testing-library";
import SelectionCard from "./SelectionCard";
import "jest-dom/extend-expect";

afterEach(cleanup);

describe("testing <SelectionCard /> component", () => {
  it("creates snapshot", () => {
    const { asFragment, rerender } = render(
      <SelectionCard
        description={"some description"}
        extraInfo={"some info"}
        iconElement={
          <i className="fas fa-chalkboard-teacher" />
        }
      >
        I am text
      </SelectionCard>,
    );
    expect(asFragment()).toMatchSnapshot();

    rerender(
      <SelectionCard
        description={"some description"}
        iconElement={
          <i className="fas fa-chalkboard-teacher" />
        }
      >
        I am text
      </SelectionCard>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
