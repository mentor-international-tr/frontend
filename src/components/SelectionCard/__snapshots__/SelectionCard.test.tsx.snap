// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <SelectionCard /> component creates snapshot 1`] = `
<DocumentFragment>
  <button
    class="selection-card"
  >
    <div
      class="selection-card__icon-container"
    >
      <i
        class="fas fa-chalkboard-teacher"
        data-testid="main-icon"
      />
    </div>
    <h2
      class="selection-card__description"
    >
      some description
    </h2>
    <p
      class="selection-card__extra-info"
    >
      some info
    </p>
  </button>
</DocumentFragment>
`;

exports[`testing <SelectionCard /> component creates snapshot 2`] = `
<DocumentFragment>
  <button
    class="selection-card"
  >
    <div
      class="selection-card__icon-container"
    >
      <i
        class="fas fa-chalkboard-teacher"
        data-testid="main-icon"
      />
    </div>
    <h2
      class="selection-card__description"
    >
      some description
    </h2>
  </button>
</DocumentFragment>
`;
